frontend/packages/pipelines-plugin/src/utils/pipeline-utils.ts

```diff
@@ -129,9 +129,11 @@ export const PipelineResourceListFilterLabels = {
  * Appends the pipeline run status to each tasks in the pipeline.
  * @param pipeline
  * @param pipelineRun
+ * @param isFinallyTasks
  */
-export const appendPipelineRunStatus = (pipeline, pipelineRun) => {
-  return _.map(pipeline.spec.tasks, (task) => {
+export const appendPipelineRunStatus = (pipeline, pipelineRun, isFinallyTasks = false) => {
+  const tasks = isFinallyTasks ? pipeline.spec.finally : pipeline.spec.tasks;
```

Comment:
If you call this function directly with `isFinallyTasks` param set to true but with a pipeline that doesn't contain the finally task. This will throw error so please add this fix and also add this assertion to your test case as well.

```suggestion
const tasks =(isFinallyTasks ? pipeline.spec?.finally : pipeline.spec.tasks) || []
```

Reply:
could we return `[ ]` when tasks is undefined or empty instead of going further into the function?

Reply:
we could also go with the suggested change and let `map` handle the rest.